interp/src/interpreter/interpret_group.rs

```diff
@@ -37,13 +37,16 @@ fn interp_assignments<'a, I: Iterator<Item = &'a ir::Assignment>>(
 
     let cells = get_cells(assigns.iter().copied());
 
-    let possible_ports: HashSet<*const ir::Port> =
-        assigns.iter().map(|a| get_const_from_rrc(&a.dst)).collect();
+    let mut possible_ports: HashSet<PortAssignment> = HashSet::new();
+    for assign in &assigns {
+        let pa = PortAssignment::new(&*assign.dst.borrow(), assign);
+        possible_ports.insert(pa);
+    }
```

Comment:
Maybe a silly question, but any particular reason why this needed to use a `for` loop? It seems intuitively like `map(...).collect()` could still work, like `.map(|a| PortAssignment::new(&*a.dst.borrow(), a))`, but maybe that creates weird lifetime problems.

Reply:
Sorry, you are right. I forgot to change that back in I guess.